lib/HLSL/HLModule.cpp

```diff
@@ -1220,6 +1236,13 @@ void HLModule::CreateElementGlobalVariableDebugInfo(
 
   DIType *DITy = DIGV->getType().resolve(EmptyMap);
   DIScope *DITyScope = DITy->getScope().resolve(EmptyMap);
+
+  // If element size is greater than base size for some reason (for example: 
+  // empty struct), just cap it at that size.
+  unsigned compositeSize = GetCompositeTypeSize(DITy);
```

Comment:
Could we have a fix that's a bit more targeted to the empty struct case? I feel like this should be an assert in the general case and not just silently truncate.

Reply:
Also, this truncation logic is incorrect if `offsetInBits > 0`

Reply:
I don't expect the offset in bits to be > 0. Added an assert.